### PR TITLE
Added new unit: triple shot tank

### PR DESCRIPTION
**Pull request by the-bucket-rider**
Created a new unit:  triple shot tank. This new tank periodically shoots three consecutive bullets when firing.

### CODE REVIEW

File: src/battle_game/core/units/triple_shot_star_tank.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include "battle_game/core/unit.h"
+
+namespace battle_game::unit {
+class Triple_shot_star_tank : public Unit {
```
Comment (LazyJazz) [inline src/battle_game/core/units/triple_shot_star_tank.h:5]: Class name should be in CamelCase

**Comment by the-bucket-rider** [comment]
屏幕截图(1)

File: src/battle_game/core/units/triple_shot_star_tank.cpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  if (player) {
+    auto &input_data = player->GetInputData();
+    auto diff = input_data.mouse_cursor_position - position_;
+    if (glm::length(diff) < 1e-4) {
+      turret_rotation_ = rotation_;
+    } else {
+      turret_rotation_ = std::atan2(diff.y, diff.x) - glm::radians(90.0f);
+    }
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/triple_shot_star_tank.cpp:131]: Inherit from unit::Tank if you are reusing these function

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/triple_shot_star_tank.cpp (hunk 2 above)
Comment (the-bucket-rider) [reply src/battle_game/core/units/triple_shot_star_tank.cpp]: I have modified my code. The new tank class now inherits from unit::Tank.

File: src/battle_game/core/units/triple_shot_star_tank.h (hunk 1 above)
Comment (the-bucket-rider) [reply src/battle_game/core/units/triple_shot_star_tank.h]: Modified class name as requested.

**Comment by the-bucket-rider** [comment]
Modefied code as requested.

**Review by LazyJazz — DISMISSED** [review]
LGTM!

**Pull request merged**